lint-workflow/lint.py

```diff
@@ -3,79 +3,189 @@
 import yaml
 import json
 import urllib3 as urllib
+import sys
+from urllib3.util import Retry
+from urllib3.exceptions import MaxRetryError
+import logging
+
+PROBLEM_LEVELS = {
+    'warning': 1,
+    'error': 2,
+}
+
+class Colors:
+    """Class containing color codes for printing strings to output."""
+    black = "30m"
+    red = "31m"
+    green = "32m"
+    yellow = "33m"
+    blue = "34m"
+    magenta = "35m"
+    cyan = "36m"
+    white = "37m"
+
+class LintFinding(object):
+    """Represents a linting problem."""
+    def __init__(self, description='<no description>', level=None):
+        self.description = description
+        self.level = level
+
+
+def get_error_level(findings):
+    """Get max error level from list of findings."""
+    max_error_level = 0
+
+    for finding in findings:
+        max_error_level = max(max_error_level, PROBLEM_LEVELS[finding.level])
+
+    return max_error_level
+
+def print_finding(finding: LintFinding):
+    """Print formatted and colored finding."""
+    if finding.level == 'warning':
+        color = Colors.yellow
+    elif finding.level == 'error':
+        color = Colors.red
+    else:
+        color = Colors.white
+
+    line = f'  - \033[{color}{finding.level}\033[0m '
+    # line += max(38 - len(line), 0) * ' '
+    line += finding.description
+
+    print(line)
+
+def get_github_api_response(url, action_id):
```

Comment:
Exceptions handling no. But the retries can increase the time (since there are three retirees, and every retry has more delay between the previous and itself). So My thought if we are hitting the GH  API limit on every run, we can skip the retry, and leave only the exception handling.

Reply:
I think we should try running this on a workflow that has a lot of actions to check to make sure that the time it takes for the retries isn't too long. Maybe a flag to turn on retries from the cli could be passed in if we want them?

Reply:
I think the time for retries is not worth it, as we spoke yesterday.We can come back to this idea in the future I think :)